api/keto/acl/v1/check_service.proto

```diff
@@ -0,0 +1,71 @@
+syntax = "proto3";
+
+package keto.acl.v1;
+
+import "keto/acl/v1/acl.proto";
+import "google/api/client.proto";
+
+option go_package = "github.com/ory/keto/api/keto/acl/v1;acl";
+option csharp_namespace = "Ory.Keto.Acl.V1";
+option java_multiple_files = true;
+option java_outer_classname = "CheckServiceProto";
+option java_package = "sh.ory.keto.acl.v1";
+option php_namespace = "Ory\\Keto\\Acl\\V1";
+
+// The service that performs authorization checks
+// based on the stored Access Control Lists.
+service CheckService {
+  option (google.api.default_host) = "keto.exampleapis.com";
+  // Check performs an authorization check.
+  rpc Check(CheckRequest) returns (CheckResponse);
+}
+
+// The request for a CheckService.Check rpc.
+// Checks whether a specific subject is related to an object.
+message CheckRequest {
+  // The related object in this check.
+  Object object = 1;
+  // The relation this check.
+  string relation = 2;
+  // The concrete subject id to check.
+  string subject_id = 3;
+  // Optional. The staleness bound of this check.
+  //
+  // It specifies the clients last known snapshot token
+  // that the server uses to only check against ACLs newer
+  // than specified token snapshot in order to compute the
+  // result of this check.
+  //
+  // Leave this field blank if...
+  //  - your application strictly requires to act on up-to-date data
+  //  - your application not yet has a snaptoken on the related data (e.g. user)
+  //
+  // If the specified token is too old, the server falls back and tries to
+  // find the best snapshot token to perform the check using least stale
+  // ACLs that very likely already got replicated.
+  //
+  // *It is recommended to perform checks using slightly stale
+  // data (e.g. token older than 3-10 seconds) for minimum latency
+  // and where the application is allowed to accept slightly off checks.
+  bytes snaptoken = 4;
```

Comment:
Maybe it would be better to add them once we have them in the actual code? As they are not strictly required, we can first implement everything without snaptokens/zookies/... and add them in the next step? Do you think it should be in the API definition already anyway?

Reply:
Yes, I do think having them in the API right from the beginning is better.
Users need to learn and adopt the tokens beforehand so they don't need to rewrite their code logics too much.

For now they do not have any special effect since the server always uses the newest tuples.
We could make the server always return a `coming_soon` snaptoken for now. Not an empty one!

Reply:
That is a good idea